distgit/upload.py

```diff
@@ -0,0 +1,103 @@
+import os
```

Comment:
There is one unpleasant aspect about this change that it makes the functions below basically a public interface. That might not be desired though because they shouldn't be changed from now on in backward incompatible manner (unless bumping major). Can you either add comment into the file that this is private and suspectible to change or alternatively prefix everything with `__` ? What do you think?

Reply:
That's a valid point, but in my opinion, we want some middle ground from what you are proposing.

> it makes the functions below basically a public interface. That might not be desired though because they shouldn't be changed from now on in backward incompatible manner (unless bumping major).

Not particularly. There is no statement or promise, that this is a stable API, so we are not encouraging anyone (who doesn't know what he is doing) to use those functions. I see it the same way as if somebody wanted to use e.g. whatever function from `rpkglib/utils.py` (from rpkg-util project). Those are not really a public interface.

> Can you either add comment into the file that this is private and suspectible to change

I can do that but see the paragraph above. As I understand it, everything is implicitly private unless someone promises differently.

> or alternatively prefix everything with `__` ?

This is what I don't want to do because I actually want to use those functions from somewhere else. Let me elaborate.

_We_ have a downstream distgit package that _duplicates_ a lot of code. My end goal is to move all the common parts here (or to some common library) and then just import them in the downstream package. The benefits are obvious for both parties. There is a lot of work to be done and we decided to proceed in small steps so we can be sure that both upstream and downstream are happy with the changes. Since it will be an iterative process, I am afraid that having the possibility to break things will be necessary. At least from the beginning until we figure out, how the common API could even look like. At this point, we have no idea, so having the possibility to experiment is IMHO essential. But I think, it's just fine because we are migrating only one downstream project and we are able to communicate the incompatible changes and provide patches for them.

Reply:
I think at least comment should be added that this stuff is not a public interface and suspect to change.

I am ok when somebody will use it anyway (e.g. you) but the person should know explicitly that this can change at any time e.g. by me or by you.

Btw. can you show me the downstream distgit package so that I know what you actually want to do?

Reply:
> I think at least comment should be added that this stuff is not a public interface and suspect to change.

I will do that. There is no hurt in explicitly stating that.

Reply:
> Btw. can you show me the downstream distgit package so that I know what you actually want to do?

I will see what I can do

Reply:
The important thing is whether the new methods are or aren't meant
to be API, case by case.  Only those which are should be moved.

Those which are going to be moved, are API.   I see no reason to
document it as unstable :-) that's just wasting of time.  We know the
users, and we don't release it yet.